galerautils/src/gu_enc_utils.cpp

```diff
@@ -0,0 +1,163 @@
+      log_error << "Failed to generate random key using SSL.";
+      for (size_t i = 0; i < Aes_ctr::FILE_KEY_LENGTH; ++i) {
+        buf[i] = rand() % 255;
+      }
+    }
+    return std::string(buf, Aes_ctr::FILE_KEY_LENGTH);
+}
+
+static unsigned char iv[gu::Aes_ctr_decryptor::AES_BLOCK_SIZE] = {0};
+std::string EncryptKey(const std::string &key_to_be_encrypted, const std::string &key)
+{
+    if (key_to_be_encrypted.length() != Aes_ctr::FILE_KEY_LENGTH ||
+        key.length() != Aes_ctr::FILE_KEY_LENGTH) {
+        log_fatal << "Encryption key length mismatch."
+                    << " key_to_be_encrypted.length(): " << key_to_be_encrypted.length()
+                    << " key.length(): " << key.length()
+                    << " Expected: " << Aes_ctr::FILE_KEY_LENGTH
+                    << " Aborting.";
+        abort();
+    }
+
+    const unsigned char* key_ptr = reinterpret_cast<const unsigned char*>(key.c_str());
+    const unsigned char* key_to_be_encrypted_ptr = reinterpret_cast<const unsigned char*>(key_to_be_encrypted.c_str());
+    char result_buf[gu::Aes_ctr::FILE_KEY_LENGTH];
+    unsigned char* result_buf_ptr = reinterpret_cast<unsigned char*>(result_buf);
+
+    Aes_ctr_encryptor encryptor;
+    encryptor.open(key_ptr, iv);
+    encryptor.encrypt(result_buf_ptr, key_to_be_encrypted_ptr, key_to_be_encrypted.length());
+    encryptor.close();
+
+    return std::string(result_buf, gu::Aes_ctr::FILE_KEY_LENGTH);
+}
+
+std::string DecryptKey(const std::string &key_to_be_decrypted, const std::string &key)
+{
+    if (key_to_be_decrypted.length() != Aes_ctr::FILE_KEY_LENGTH ||
+        key.length() != Aes_ctr::FILE_KEY_LENGTH) {
+        log_fatal << "Encryption key length mismatch."
+                    << " key_to_be_decrypted.length(): " << key_to_be_decrypted.length()
+                    << " key.length(): " << key.length()
+                    << " Expected: " << Aes_ctr::FILE_KEY_LENGTH
+                    << " Aborting.";
+        abort();
+    }
+
+    const unsigned char* key_ptr = reinterpret_cast<const unsigned char*>(key.c_str());
+    const unsigned char* key_to_be_decrypted_ptr = reinterpret_cast<const unsigned char*>(key_to_be_decrypted.c_str());
+    char result_buf[gu::Aes_ctr::FILE_KEY_LENGTH];
+    unsigned char* result_buf_ptr = reinterpret_cast<unsigned char*>(result_buf);
+
+    Aes_ctr_decryptor decryptor;
+    decryptor.open(key_ptr, iv);
+    decryptor.decrypt(result_buf_ptr, key_to_be_decrypted_ptr, key_to_be_decrypted.length());
+    decryptor.close();
+    return std::string(result_buf, gu::Aes_ctr_decryptor::FILE_KEY_LENGTH);
+}
+
+std::string CreateMasterKeyName(const UUID& const_uuid, const UUID& uuid, int keyId) {
+    static const std::string MASTER_KEY_PREFIX = "GaleraKey-";
+    static const std::string MASTER_KEY_SEQNO_SEPARATOR = "-";
+    static const std::string MASTER_KEY_ID_SEPARATOR = "@";
+    std::ostringstream os;
+    os << uuid << MASTER_KEY_ID_SEPARATOR << const_uuid;
```

Comment:
This is just a formatting question, but why does this function mix string construction with << and +? It would be cleaner to use the stringstream for everything, and then the static const could be plain const char*s.

Reply:
Done